controllers/dockercluster_controller.go

```diff
@@ -39,28 +40,29 @@ type DockerClusterReconciler struct {
 func (r *DockerClusterReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 	ctx := context.Background()
 	log := r.Log.WithValues("dockercluster", req.NamespacedName)
+	log.Info("Reconciling cluster")
 
 	dockerCluster := &infrastructurev1alpha2.DockerCluster{}
 	if err := r.Client.Get(ctx, req.NamespacedName, dockerCluster); err != nil {
 		if apierrors.IsNotFound(err) {
 			return ctrl.Result{}, nil
 		}
-		log.Error(err, "failed to get dockerMachine")
+		log.Error(err, "failed to get cluster")
 		return ctrl.Result{}, err
 	}
 
 	if dockerCluster.Status.Ready {
 		return ctrl.Result{}, nil
 	}
 
+	if err := r.reconcileNetwork(dockerCluster.Name, r.Log); err != nil {
+		log.Error(err, "Failed to reconcile network for cluster")
+		return ctrl.Result{}, err
+	}
+
 	// Store Config's state, pre-modifications, to allow patching
 	patchCluster := client.MergeFrom(dockerCluster.DeepCopy())
```

Comment:
This should likely be moved up earlier in the `Reconcile` method to avoid inadvertent changes to `dockerCluster` that might not be persisted.

Reply:
Fixed.